HeterogeneousCore/SonicTriton/src/TritonService.cc

```diff
@@ -0,0 +1,296 @@
+#include "HeterogeneousCore/SonicTriton/interface/TritonService.h"
+#include "HeterogeneousCore/SonicTriton/interface/triton_utils.h"
+
+#include "DataFormats/Provenance/interface/ModuleDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/ServiceRegistry/interface/ProcessContext.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "grpc_client.h"
+#include "grpc_service.pb.h"
+
+#include <cstdio>
+#include <cstdlib>
+#include <filesystem>
+#include <utility>
+#include <tuple>
+
+namespace ni = nvidia::inferenceserver;
+namespace nic = ni::client;
+
+const std::string TritonService::Server::fallbackName{"fallback"};
+const std::string TritonService::Server::fallbackAddress{"0.0.0.0"};
+
+namespace {
+  std::pair<std::string, int> execSys(const std::string& cmd) {
+    //redirect stderr to stdout
+    auto pipe = popen((cmd + " 2>&1").c_str(), "r");
+    int thisErrno = errno;
+    if (!pipe)
+      throw cms::Exception("SystemError") << "popen() failed with errno " << thisErrno << " for command: " << cmd;
+
+    //extract output
+    constexpr static unsigned buffSize = 128;
+    std::array<char, buffSize> buffer;
+    std::string result;
+    while (!feof(pipe)) {
+      if (fgets(buffer.data(), buffSize, pipe))
+        result += buffer.data();
+      else {
+        thisErrno = ferror(pipe);
+        if (thisErrno)
+          throw cms::Exception("SystemError") << "failed reading command output with errno " << thisErrno;
+      }
+    }
+
+    int rv = pclose(pipe);
+    return std::make_pair(result, rv);
+  }
+
+  std::string getUuid() {
+    auto [uuid, rv] = execSys("uuidgen | sed 's/-//g'");
+    uuid.pop_back();
+    return uuid;
+  }
```

Comment:
How about calling
https://github.com/cms-sw/cmssw/blob/5fbd6f99995d8082f4c594d039322704591beea5/FWCore/Utilities/interface/GlobalIdentifier.h#L6
instead?
(it uses `uuid_generate()` from libuuid internally)